kasa/aesprotocol.py

```diff
@@ -0,0 +1,497 @@
+"""Implementation of the TP-Link AES Protocol.
+
+Based on the work of https://github.com/petretiandrea/plugp100
+under compatible GNU GPL3 license.
+"""
+
+import asyncio
+import base64
+import hashlib
+import logging
+import time
+import uuid
+from pprint import pformat as pf
+from typing import Dict, Optional, Union
+
+import httpx
+from cryptography.hazmat.primitives import hashes, padding, serialization
+from cryptography.hazmat.primitives.asymmetric import padding as asymmetric_padding
+from cryptography.hazmat.primitives.asymmetric import rsa
+from cryptography.hazmat.primitives.ciphers import Cipher, algorithms, modes
+
+from .credentials import Credentials
+from .exceptions import AuthenticationException, SmartDeviceException
+from .json import dumps as json_dumps
+from .json import loads as json_loads
+from .protocol import TPLinkProtocol
+
+_LOGGER = logging.getLogger(__name__)
+logging.getLogger("httpx").propagate = False
+
+
+def _md5(payload: bytes) -> bytes:
+    digest = hashes.Hash(hashes.MD5())  # noqa: S303
+    digest.update(payload)
+    hash = digest.finalize()
+    return hash
+
+
+def _sha1(payload: bytes) -> str:
+    sha1_algo = hashlib.sha1()  # noqa: S324
+    sha1_algo.update(payload)
+    return sha1_algo.hexdigest()
+
+
+class TPLinkAes(TPLinkProtocol):
+    """Implementation of the AES encryption protocol.
+
+    AES is the name used in device discovery for TP-Link's TAPO encryption
+    protocol, sometimes used by newer firmware versions on kasa devices.
+    """
+
+    DEFAULT_PORT = 80
+    DEFAULT_TIMEOUT = 5
+    DISCOVERY_QUERY = {"system": {"get_sysinfo": None}}
+    KASA_SETUP_EMAIL = "kasa@tp-link.net"
+    KASA_SETUP_PASSWORD = "kasaSetup"  # noqa: S105
```

Comment:
These are not currently used at all, so maybe remove them for now.

Reply:
Done